### PR TITLE
add client tracing interfaces

### PR DESCRIPTION
**Pull request by Zach41**
add interfaces to pass tracing variables:
- log level
- client trace id
ISSUE: #307

### CODE REVIEW

File: client/ctx.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"context"
+
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	logLevelRPCMetaKey = "log_level"
+	clientRequestIDKey = "client_request_id"
```
Comment (congqixia) [inline client/ctx.go:11]: It is recommended to define context key type other than using const string

File: client/ctx.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	return metadata.AppendToOutgoingContext(ctx, logLevelRPCMetaKey, infoLevel)
+}
+
+func WithWarnLogLevel(ctx context.Context) context.Context {
+	return metadata.AppendToOutgoingContext(ctx, logLevelRPCMetaKey, warnLevel)
+}
+
+func WithErrorLogLevel(ctx context.Context) context.Context {
+	return metadata.AppendToOutgoingContext(ctx, logLevelRPCMetaKey, errorLevel)
+}
```
Comment (congqixia) [inline client/ctx.go:33]: Maybe a unified method like WithLogLevel(ctx, logLevel) will do

File: client/ctx.go (hunk 1 above)
Comment (Zach41) [reply client/ctx.go:11]: These are metadata keys for grpc context, it's required to use string as key

File: client/ctx.go (hunk 2 above)
Comment (Zach41) [reply client/ctx.go:33]: A unified interface is little more flexible，`logLevel` can be any string, may cause undefined problems

File: client/ctx.go (hunk 1 above)
Comment (congqixia) [reply client/ctx.go:11]: ok

File: client/ctx.go (hunk 2 above)
Comment (congqixia) [reply client/ctx.go:33]: Undefined log level problem could be handled with parameter validation. Anyway, this implement could work.

**Pull request merged**